### PR TITLE
Update spirv-headers to latest commit before spirv-tools 2021.1 release date

### PR DESCRIPTION
**Pull request by espositofulvio**
Specify library name and version:  **spirv-headers/cci.20210414**

This pull requests update the spirv-headers package to the latest commit before spirv-tools 2021.1 release. The commit id is dafead1765f6c1a5f9f8a76387dcb2abe4e54acd and I checked that the spirv-tools v2021.1 builds with this update (the update to other packages will be pushed in separate pull requests).

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/spirv-headers/all/conanfile.py [hunk 1]
```diff
@@ -30,6 +31,8 @@ def source(self):
         extracted_dir = "SPIRV-Headers-" + self.version
         if self.version == "1.5.1":
             extracted_dir = extracted_dir + ".corrected"
+        if self.version.startswith('cci.'):
+            extracted_dir = glob.glob(self.name + "-*/")[0]
         os.rename(extracted_dir, self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/spirv-headers/all/conanfile.py:36]: `glob` can be avoided, and `source()`simplified with:

```python
def source(self):
    tools.get(**self.conan_data["sources"][self.version],
              destination=self._source_subfolder, strip_root=True)
```

Comment (espositofulvio) [reply recipes/spirv-headers/all/conanfile.py]: Done, thanks :)

File: recipes/spirv-headers/all/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 from conans import ConanFile, tools, CMake
 import os
+import glob
```
Comment (SpaceIm) [inline recipes/spirv-headers/all/conanfile.py:3]:
```suggestion
```

useless now

Comment (espositofulvio) [reply recipes/spirv-headers/all/conanfile.py]: Well I was quite sure I removed that, that's why it's in the commit message. Anyway, fixed.

**Comment by espositofulvio** [comment]
@SpaceIm I'm trying to figure out the error in building the package (I tested it on Windows with MSVC). I guess the problem is to do with the install step in Spirv-headers CMakeLists.txt:

    if (SPIRV_HEADERS_ENABLE_INSTALL)
        message(STATUS "Installing SPIRV-Header")

        set(config_install_dir "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
        ...
        install(
            FILES "${project_config}" "${version_config}"
            DESTINATION "${config_install_dir}"
        )
        ...


but am not sure how to fix this. It looks like only the cmake config files end up in the share/ folder. In this case is it ok to just remove the install command for cmake config? I could write a patch for that or use tools.replace_in_file.

**Comment by espositofulvio** [comment]
> @SpaceIm I'm trying to figure out the error in building the package (I tested it on Windows with MSVC). I guess the problem is to do with the install step in Spirv-headers CMakeLists.txt:
> 
> ```
> if (SPIRV_HEADERS_ENABLE_INSTALL)
>     message(STATUS "Installing SPIRV-Header")
> 
>     set(config_install_dir "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
>     ...
>     install(
>         FILES "${project_config}" "${version_config}"
>         DESTINATION "${config_install_dir}"
>     )
>     ...
> ```
> 
> but am not sure how to fix this. It looks like only the cmake config files end up in the share/ folder. In this case is it ok to just remove the install command for cmake config? I could write a patch for that or use tools.replace_in_file.

As an additional note, it looks like that in one of the latest commits they changed 

>     set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")

to

>     set(config_install_dir "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")

And that's what's causing the issue with the "share" folders. Also, it looks like .cmake files are not allowed, even though they were previously included, so I'm not sure if hook was recently added.

Any guidance here?

**Comment by SpaceIm** [comment]
cmake files were not included previously in this recipe, they were removed by `tools.rmdir(os.path.join(self.package_folder, "lib"))`. Add `tools.rmdir(os.path.join(self.package_folder, "share"))` below this line please.

**Review by SpaceIm — APPROVED** [review]
What a mess that spirv-headers maintainers didn't push a new tag since last November, and that most Vulkan projects rely on some random commits of spirv-headers instead of official releases.

**Review by uilianries — APPROVED** [review]

**Review by AndreyMlashkin — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**